src/control/server/harness.go

```diff
@@ -299,11 +299,12 @@ func (h *IOServerHarness) Start(parent context.Context, membership *system.Membe
 	defer shutdown()
 
 	for {
-		// Single daos_server dRPC server to handle all iosrv requests
-		if err := drpcSetup(ctx, h.log, cfg.SocketDir, h.Instances(), cfg.TransportConfig); err != nil {
-			return errors.WithMessage(err, "dRPC setup")
+		if cfg != nil {
+			// Single daos_server dRPC server to handle all iosrv requests
+			if err := drpcSetup(ctx, h.log, cfg.SocketDir, h.Instances(), cfg.TransportConfig); err != nil {
+				return errors.WithMessage(err, "dRPC setup")
+			}
```

Comment:
I couldn't see a sensible way of mocking drpcSetup without making significant changes that seem outside the scope of this PR to me, this felt like an acceptable compromise for the moment.

Reply:
I will revisit this when adding harness tests for system stop and start

Reply:
OK. For future reference, I'm fine with TODOs as long as they're documented, e.g.
```go
// TODO (DAOS-xxxx): Update tests rather than changing production code
```